extensions/vertx/deployment/src/main/java/io/quarkus/vertx/deployment/VertxProcessor.java

```diff
@@ -114,7 +115,14 @@ void collectEventConsumers(
                 if (consumeEvent != null) {
                     // Validate method params and return type
                     List<Type> params = method.parameters();
-                    if (params.size() != 1) {
+                    if (params.size() == 2) {
+                        if (!isMessageHeaders(params.get(0).name())) {
+                            // If there are two parameters, the first must be message headers.
```

Comment:
I think that we should either validate that the second param is not `Message` or modify the `EventBusConsumer` appropriately.  My guess is that `@ConsumeEvent void consume(MultiMap headers, Message<String> message)` would currently fail at runtime...

Reply:
We would need to verify with test all the supported parameter permutations (and document them)

Reply:
Hm, if we require the headers to be always the first param (current behavior) then the tests are in place, except for validation. And yes, we need to document this better - see my comment https://github.com/quarkusio/quarkus/pull/21653#discussion_r756659303.

Reply:
@mkouba I've added a check to handle `@ConsumeEvent void consume(MultiMap headers, Message<String> message)`, and updated the JavaDoc for `@ConsumeEvent` to explicitly specify valid parameters.